File: src/client/java/net/legitimoose/bot/discord/command/FindCommand.java
package net.legitimoose.bot.discord.command;

import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import net.legitimoose.bot.chat.GameChatHandler;
import net.legitimoose.bot.util.DiscordUtil;
import net.legitimoose.bot.util.McUtil;
import net.minecraft.client.Minecraft;

import java.util.concurrent.TimeUnit;

import static net.legitimoose.bot.LegitimooseBot.LOGGER;

public class FindCommand extends ListenerAdapter {
    @Override
    public void onSlashCommandInteraction(SlashCommandInteractionEvent event) {
        if (!event.getName().equals("find")) return;
        String player = event.getOption("player").getAsString();
        if (!McUtil.isValidMCUsername(player)) {
            event.reply("player name too long, sorry!").setEphemeral(true).queue();
            return;
        }
        Minecraft.getInstance().player.connection.sendCommand(McUtil.sanitizeString("find " + player));
        try {
            TimeUnit.SECONDS.sleep(1);
        } catch (InterruptedException e) {
            LOGGER.error(e.getMessage());
        }
        event.reply(DiscordUtil.sanitizeString(GameChatHandler.getInstance().lastMessages.getLast().replace(" Click HERE to join.", "").trim())).queue();
    }
}

File: src/client/kotlin/net/legitimoose/bot/discord/FindCommand.kt
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.fabricmc.fabric.api.client.message.v1.ClientReceiveMessageEvents
import net.minecraft.client.Minecraft
import net.minecraft.network.chat.Component

class FindCommand(override val event: SlashCommandInteractionEvent, val player: String) : Command {
    override fun onCommandReceived() {
        if (player.length >= 200) {
            event.reply("player name too long, sorry!").setEphemeral(true).queue()
            return
        }
        Minecraft.getInstance().player?.connection?.sendCommand("find " + player.replace("§", "?"))
        val bool: BooleanArray = booleanArrayOf(true)
        ClientReceiveMessageEvents.GAME.register { message: Component, _: Boolean ->
            if (!bool[0]) return@register
            event.reply(message.string.replace(" Click HERE to join.", "").trim()).queue()
            bool[0] = false
        }
    }
}
